### PR TITLE
MULE-8922: Enricher broken when target expression references a key in…

### PR DESCRIPTION
**Pull request by pablokraan**
…side a map

### CODE REVIEW

File: core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java [hunk 1]
```diff
@@ -53,16 +57,42 @@ protected void doAssignmentDataTypePropagationTest(PropertyScope scope, String e
...
+        assertThat(testEvent.getMessage().getPropertyDataType(PROPERTY_NAME, scope), like(Map.class, UNKNOWN, CUSTOM_ENCODING));
+    }
+
+    private CompiledExpression compileMelExpression(String expression, MuleEvent testEvent)
+    {
         final ParserConfiguration parserConfiguration = MVELExpressionLanguage.createParserConfiguration(Collections.EMPTY_MAP);
         final MVELExpressionLanguageContext context = createMvelExpressionLanguageContext(testEvent, parserConfiguration);
 
         CompiledExpression compiledExpression = (CompiledExpression) compileExpression(expression, new ParserContext(parserConfiguration));
+
```
Comment (marianogonzalez) [inline core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java]: remove unnecessary line

File: core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java [hunk 2]
```diff
@@ -53,16 +57,42 @@ protected void doAssignmentDataTypePropagationTest(PropertyScope scope, String e
...
+        MuleEvent testEvent = getTestEvent(TEST_MESSAGE);
+        final Map<String, String> propertyValue = new HashMap<>();
+        propertyValue.put(INNER_PROPERTY_NAME, TEST_MESSAGE);
+        testEvent.getMessage().setProperty(PROPERTY_NAME, propertyValue, scope, expectedDataType);
+
+        CompiledExpression compiledExpression = compileMelExpression(expression, testEvent);
+
+        // Attempts to propagate a different dataType, which should be ignored
+        dataTypePropagator.propagate(testEvent.getMessage(), new TypedValue(propertyValue, DataType.STRING_DATA_TYPE), compiledExpression);
```
Comment (marianogonzalez) [inline core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java]: I might be wrong but this looks weird to me:

* is it ok to ignore a bad datatype? should it fail instead?
* I don't know all the "functionals" about this, but it seems to me like you're testing several things on the same method. Maybe this should be splitted into several tests?

Comment (pablokraan) [inline core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java]: There are two datatypes involved. One is the map dataType which is type of the property in the message. Another one is the String dataType which is the dataType of the value which is enriched with the map expression.
This should not fail at all, is just checking that the propagation didn't affect the dataType in the property

File: core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java (hunk 1 above)
Comment (pablokraan) [inline core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java]: All empty lines are unnecessary from the code point of view, is really interesting to me to know what makes this line "unnecesary" for you. For me, this line separates parts of the test setup code to make it more readeable (for me at least).

Comment (marianogonzalez) [inline core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java]: github's diff view gave me the impression there already was an empty line. Disregard

**Pull request merged**